File: javascript/fizzbuzzify-string.test.js

const expect = require('chai').expect
const fizzbuzzify = require('./fizzbuzzify')

describe('fizzbuzzify', () => {

  it("should convert a string to a fizzbuzzified string.", () => {
    expect(fizzbuzzify('')).to.equal('')
    expect(fizzbuzzify('ok')).to.equal('ok')
    expect(fizzbuzzify('ok ok ok')).to.equal('ok ok fizz')
    expect(fizzbuzzify('ok ok ok ok ok')).to.equal('ok ok fizz ok buzz')
    expect(fizzbuzzify('ok ok ok ok ok ok ok ok ok ok ok ok ok ok ok')).to.equal(
      'ok ok fizz ok buzz fizz ok ok fizz buzz ok fizz ok ok fizzbuzz')

  })

})

File: clojure/Fizzbuzz/fizzbuzzify_string_test.clj
(ns Fizzbuzzify.fizzbuzzify_string_test
  (require [Fizzbuzz.fizzbuzzify_string :refer [fizzbuzzify]]))
(use 'clojure.test)

(deftest fizzbuzzifyTest
  "should convert a string to a fizzbuzzified string."
  (is (= (fizzbuzzify "") "d"))
  (is (= (fizzbuzzify "ok") "ok"))
  (is (= (fizzbuzzify "ok ok ok") "ok ok fizz"))
  (is (= (fizzbuzzify "ok ok ok ok ok") "ok ok fizz ok buzz"))
  (is (= (fizzbuzzify "ok ok ok ok ok ok ok ok ok ok ok ok ok ok ok") 
         "ok ok fizz ok buzz fizz ok ok fizz buzz ok fizz ok ok fizzbuzz")))
(run-tests)